packages/alfa-rules/test/sia-r67/rule.spec.tsx

```diff
@@ -0,0 +1,112 @@
+import { Device } from "@siteimprove/alfa-device";
+import { jsx } from "@siteimprove/alfa-dom/jsx";
+import { Option } from "@siteimprove/alfa-option";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { test } from "@siteimprove/alfa-test";
+
+import { Document, Element } from "@siteimprove/alfa-dom";
+
+import R67, { Outcomes } from "../../src/sia-r67/rule";
+
+import { evaluate } from "../common/evaluate";
+import { passed, failed, inapplicable } from "../common/outcome";
+
+const { and } = Predicate;
+const { hasId } = Element;
+
+const device = Device.standard();
+
+function getElementById(document: Document): (id: string) => Element {
+  return (id) =>
+    document
+      .descendants()
+      .find(and(Element.isElement, hasId(id)))
+      .get();
+}
+
+test("evaluate() passes on elements marked as decorative with no alternative text", async (t) => {
+  const document = Document.of((self) => [
+    Element.fromElement(
+      <html>
+        <img id="empty-alt" src="foo.jpg" alt="" />
+        <video id="role-none" src="foo.mp4" role="none" />
+        <object id="role-presentation" src="foo.jpg" role="presentation" />
+        <svg id="svg" role="none">
+          <circle cx="50" cy="50" r="40" fill="yellow"></circle>
+        </svg>
+      </html>,
+      Option.of(self)
+    ),
+  ]);
+  const getById = getElementById(document);
+  const emptyAlt = getById("empty-alt");
+  const roleNone = getById("role-none");
+  const rolePresentation = getById("role-presentation");
+  const svg = getById("svg");
```

Comment:
This can be done using destructuring and without additional helpers and need for IDs:

```ts
const [img, video, object, svg] = document
  .first()
  .get()
  .children()
  .filter(Element.isElement);
```

Reply:
True, but this is less resilient to changes in the HTML.
Additionally, ids are commenting (in the HTML) what this target is testing, and can be handy for debugging messages from inside the rule itself.

Reply:
That's reasonable 👍 What we need then is a unified approach to this rather than ad-hoc helpers scattered throughout rule spec files. Test code is most annoying code to maintain so it needs to be _super_ maintainable 🙈

Reply:
Yes, I totally agree. If we decide that IDs + getter is the way we wanna go, we should of course make the helper common and update existing tests to use it…
I've also been considering writing a wrapper for `Document.of`+`Element.fromElement` that we seem to use pretty often in tests…

Reply:
#281 should hopefully help on the last point.

Reply:
Yeah it gets a bit better. Still need to 
```javascript
Document.fromDocument(
  h.document([
    <html>
      <head>
        <title>Hello world</title>
      </head>
    </html>
  ])
);
```
instead of
```javascript
documentFromHtml(<html>
      <head>
        <title>Hello world</title>
      </head>
    </html>);
```